modules/rest-compatibility/src/main/java/org/elasticsearch/rest/compat/version7/CompatibleHandlers.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.rest.compat.version7;
+
+import org.elasticsearch.common.logging.DeprecationLogger;
+import org.elasticsearch.rest.RestRequest;
+
+import java.util.function.Consumer;
+
+public class CompatibleHandlers {
+
+    public static Consumer<RestRequest> consumeParameterType(DeprecationLogger deprecationLogger) {
+        String TYPES_DEPRECATION_MESSAGE = "[types removal] Using type as a path parameter is deprecated.";
+
+        return r -> {
+            deprecationLogger.deprecatedAndMaybeLog("create_index_with_types", TYPES_DEPRECATION_MESSAGE);
```

Comment:
If I'm understanding correctly, it seems like we'll log two deprecation messages per request, both the endpoint-specific one like "Specifying types in document get requests is deprecated..." and this more general one "Using type as a path parameter is deprecated". To me we could just stick with the end-point specific one to avoid warning twice about the same issue.

As a side note, this general deprecation message is also logged under the key "create_index_with_types", I think we'd want a more precise name.

Reply:
agree - let's stick with just per endpoint warning.
WIth that I feel like we can even get rid of `CompatibleHandlers.consumeParameterType` - it would only consume a type param -  and the whole utility class. 
I was worried that it would end up a bag of all the helpers for compatibility. It might generate a little bit of duplication, but there is no need for it now.

Do you mean more precise name for a deprecation key? Like to be the same for Get and Index and other handlers?